code/game/objects/items/toys.dm

```diff
@@ -49,9 +49,14 @@
 
 /obj/item/toy/water_balloon/afterattack(atom/A, mob/user, proximity)
 	if(!proximity) return
-	if (istype(A, /obj/structure/reagent_dispensers/watertank) && get_dist(src,A) <= 1)
-		A.reagents.trans_to_obj(src, 10)
-		to_chat(user, "<span class='notice'>You fill the balloon with the contents of [A].</span>")
+	var/issink = istype(A, /obj/structure/hygiene/sink)
+	if((istype(A, /obj/structure/reagent_dispensers/watertank) || issink) && get_dist(src,A) <= 1)
+		if(!issink)
```

Comment:
I have to be honest, this is some spaghetti code with these type and if/else-checks

Reply:
I recall this being discussed on Discord and suggesting using an overrided method on /obj instead of this block.